test/promise-subclass.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = (tap) => {
+  return class SubclassedPromise extends Promise {
+    then(onSuccess, onReject) {
+      tap.type(onSuccess, 'function');
+      tap.type(onReject, 'undefined');
+      return Promise.prototype.then.call(this, onSuccess, onReject);
```

Comment:
I just checked out the branch to test it and the test passed using `super` as well 😃

I'm not sure if it's a big deal - if not, just leave it as is and feel free to merge on my account 👍